src/Autofac/Util/Cache/ReflectionCacheTupleDictionary.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
```

Comment:
Might want some independent unit tests around this.

Reply:
Added a bunch.